CNF-854 documentation updates

abhatt-rh opened the pull request:
Documentation updates for [TELCODOCS-89](https://issues.redhat.com/browse/TELCODOCS-89) and [TELCODOCS-98](https://issues.redhat.com/browse/TELCODOCS-98)

StephenJamesSmith commented on modules/cnf-performing-end-to-end-tests-for-platform-verification.adoc:
```diff
@@ -107,7 +178,16 @@ You can use the `-ginkgo.focus` parameter to filter a set of tests:
+
+You can run only the latency test using the `-ginkgo.focus` parameter.
+
+To run only the latency test, you must provide the `-ginkgo.focus` parameter and the `PERF_TEST_PROFILE` environment variable that contains the name of the performance profile need to be tested. For example:
```
To run only the latency test, you must provide the `-ginkgo.focus` parameter and the `PERF_TEST_PROFILE` environment variable that contains the name of the performance profile that needs to be tested. For example:

abhatt-rh replied:
Done.
Thank you, @StephenJamesSmith

abhatt-rh commented:
Hi team,
Request you to kindly provide you final comments for for [TELCODOCS-89](https://issues.redhat.com/browse/TELCODOCS-89) and [TELCODOCS-98](https://issues.redhat.com/browse/TELCODOCS-98)
 @MarSik @cynepco3hahue: I have incorporated the comments that you had provided the GDoc review.
 Adding @gsr-shanks for a QE review and sign-off.
 
Feel free to add more reviewers. Thank you.

vikram-redhat commented:
@abhatt-rh why is this PR removing the Gemfile?

abhatt-rh commented:
> @abhatt-rh why is this PR removing the Gemfile?

@vikram-redhat While we troubleshooting some asciibinder build and gem issue with my toolchain, you had asked me to remove it.

vikram-redhat commented:
> > @abhatt-rh why is this PR removing the Gemfile?
> 
> @vikram-redhat While we troubleshooting some asciibinder build and gem issue with my toolchain, you had asked me to remove it.

:D from the main repo? Sorry, I might have made a mistake there. In any case, this deleted file should be taken out of this commit.

gsr-shanks commented:
@mrniranjan Could you please review this PR?

mrniranjan commented:
Looks good to me.

abhatt-rh commented:
@mrniranjan Thank you!

abhatt-rh commented:
@ahardin-rh Request you to kindly review.
Here are the link to it's JIRA issues:[TELCODOCS-89](https://issues.redhat.com/browse/TELCODOCS-89) and [TELCODOCS-98](https://issues.redhat.com/browse/TELCODOCS-98)

ahardin-rh commented on modules/cnf-performing-end-to-end-tests-for-platform-verification.adoc:
```diff
@@ -24,6 +24,8 @@ Validations include:
 * Having the PTP Operator installed
 * Using OVN kubernetes as the SDN
 
+Latency tests, a part of the CNF-test container, also require the same validations. For more information about running a latency test, see link:cnf-performing-end-to-end-tests-running-the-latency_tests_{context}[Running the latency tests].
```
Unfortunately, we can't include any internal links (xrefs) in modules right now. We're hoping that xrefs can be reintroduced soon with Pantheon 2. So, you have to either have an unlinked reference to that section or remove the reference altogether. :disappointed:

ahardin-rh commented on modules/cnf-performing-end-to-end-tests-for-platform-verification.adoc:
```diff
@@ -74,18 +76,87 @@ Currently, not all tests run selectively on the nodes belonging to the pool.
 [id="cnf-performing-end-to-end-tests-running-the-tests_{context}"]
 == Running the tests
-Assuming the file is in the current folder, the command for running the test suite is:
+Assuming the kubeconfig file is in the current folder, the command for running the test suite is:
```
`kubeconfig` file

ahardin-rh commented on modules/cnf-performing-end-to-end-tests-for-platform-verification.adoc:
```diff
@@ -74,18 +76,87 @@ Currently, not all tests run selectively on the nodes belonging to the pool.
 
+[id="cnf-performing-end-to-end-tests-running-the-latency_tests_{context}"]
+=== Running the latency tests
+In OpenShift Container Platform 4.7, you can also run latency tests from the CNF-test container. The latency test allows you to set a latency limit, so that you can determine performance, throughput, and latency.
```
In {product-title} {product-version}, 

limit so that

ahardin-rh commented on modules/cnf-performing-end-to-end-tests-for-platform-verification.adoc:
```diff
@@ -74,18 +76,87 @@ Currently, not all tests run selectively on the nodes belonging to the pool.
+=== Running the latency tests
+In OpenShift Container Platform 4.7, you can also run latency tests from the CNF-test container. The latency test allows you to set a latency limit, so that you can determine performance, throughput, and latency.
+
+The latency test runs the `oslat` tool, which is an open source program to detect OS level latency. For more information, see the Red Hat Knowledgebase solution link:https://access.redhat.com/solutions/5315541[How to measure OS and hardware latency on isolated CPUs?].
```
Just FYI, It's okay to include this external link here even though we could not include an internal xref earlier in the document.

ahardin-rh commented on modules/cnf-performing-end-to-end-tests-for-platform-verification.adoc:
```diff
@@ -74,18 +76,87 @@ Currently, not all tests run selectively on the nodes belonging to the pool.
+
+The latency test runs the `oslat` tool, which is an open source program to detect OS level latency. For more information, see the Red Hat Knowledgebase solution link:https://access.redhat.com/solutions/5315541[How to measure OS and hardware latency on isolated CPUs?].
+
+By default, the latency tests are disabled. To enable the latency test, you must add the `LATENCY_TEST_RUN` variable and set its value to true. For example, `LATENCY_TEST_RUN=true`.
```
value to `true`.

ahardin-rh commented on modules/cnf-performing-end-to-end-tests-for-platform-verification.adoc:
```diff
@@ -74,18 +76,87 @@ Currently, not all tests run selectively on the nodes belonging to the pool.
+To perform the latency tests, run the following command:
+
+----
+docker run -v $(pwd)/:/kubeconfig -e KUBECONFIG=/kubeconfig/kubeconfig -e LATENCY_TEST_RUN=true -e LATENCY_TEST_RUNTIME=600 -e OSLAT_MAXIMUM_LATENCY=20 registry.redhat.io/openshift4/cnf-tests-rhel8:v4.7 /usr/bin/test-run.sh
```
Missing the preceding command prompt $

ahardin-rh commented on modules/cnf-performing-end-to-end-tests-for-platform-verification.adoc:
```diff
@@ -74,18 +76,87 @@ Currently, not all tests run selectively on the nodes belonging to the pool.
+----
+[NOTE]
+====
+You must run the latency test in link:cnf-performing-end-to-end-tests-disconnected-mode_{context}[Discovery mode].
```
We must remove this link/xref because we can't have internal xrefs in modules

ahardin-rh commented on modules/cnf-performing-end-to-end-tests-for-platform-verification.adoc:
```diff
@@ -74,18 +76,87 @@ Currently, not all tests run selectively on the nodes belonging to the pool.
+You must run the latency test in link:cnf-performing-end-to-end-tests-disconnected-mode_{context}[Discovery mode].
+====
+
+Here is an excerpt of a sample result of a 10 second latency test:
```
Instead of a "Here is" statement, you can just use this formatting to label the example:


    .Excerpt of a ample result of a 10 second latency test

ahardin-rh commented on modules/cnf-performing-end-to-end-tests-for-platform-verification.adoc:
```diff
@@ -74,18 +76,87 @@ Currently, not all tests run selectively on the nodes belonging to the pool.
+Here is an excerpt of a sample result of a 10 second latency test:
+[source,terminal]
+----
+[root@cnf12-installer ~]# podman run --rm -v $KUBECONFIG:/kubeconfig:Z -e PERF_TEST_PROFILE=worker-cnf-2 -e KUBECONFIG=/kubeconfig -e LATENCY_TEST_RUN=true -e LATENCY_TEST_RUNTIME=10 -e OSLAT_MAXIMUM_LATENCY=20 -e DISCOVERY_MODE=true registry.redhat.io/openshift4/cnf-tests-rhel8:v4.7 /usr/bin/test-run.sh
```
You'll need to separate out the command from the example output per https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output

ahardin-rh commented on modules/cnf-performing-end-to-end-tests-for-platform-verification.adoc:
```diff
@@ -107,7 +178,16 @@ You can use the `-ginkgo.focus` parameter to filter a set of tests:
+
+[source, terminal]
+----
+docker run --rm -v $KUBECONFIG:/kubeconfig -e KUBECONFIG=/kubeconfig -e LATENCY_TEST_RUN=true -e LATENCY_TEST_RUNTIME=600 -e OSLAT_MAXIMUM_LATENCY=20 -e PERF_TEST_PROFILE=<performance_profile_name> registry.redhat.io/openshift4/cnf-tests-rhel8:v4.7 /usr/bin/test-run.sh -ginkgo.focus="\[performance\]\[config\]|\[performance\]\ Latency\ Test"
```
$ docker run

ahardin-rh commented:
@abhatt-rh Just a few minor style edits from me! Please let me know if you have any questions. Thanks and great job!

abhatt-rh commented:
@ahardin-rh Thank you for the peer review and for all the help with my queries. I have addressed all the comments and committed the changes. Let me know if I can go ahead and stash the commits for merge.

ahardin-rh commented on modules/cnf-performing-end-to-end-tests-for-platform-verification.adoc:
```diff
@@ -74,18 +76,91 @@ Currently, not all tests run selectively on the nodes belonging to the pool.
+You must run the latency test in discovery mode. For more information, see the Discovery mode section.
+====
+
+ .Excerpt of a sample result of a 10-second latency test using the following command:
```
https://cnf-854-perf-tests-docs--ocpdocs.netlify.app/openshift-enterprise/latest/scalability_and_performance/cnf-performance-addon-operator-for-low-latency-nodes.html#cnf-performing-end-to-end-tests-running-the-latency_tests_cnf-master

In the preview build, it looks like the lead-in titles for .Excerpt...   and   .Example output are rendering as their own code blocks. I think there are some extra spaces before each that might be throwing the code off.

Everything else looks great!

ahardin-rh commented:
Looks great! Merging!

ahardin-rh commented:
/cherrypick enterprise-4.7
